src/ruby.ts

```diff
@@ -62,27 +62,10 @@ export class Ruby {
   }
 
   private async activate(ruby: string) {
-    let shellProfilePath;
-    // eslint-disable-next-line no-process-env
-    const shell = process.env.SHELL?.split("/").pop();
-    // eslint-disable-next-line no-process-env
-    const home = process.env.HOME;
-
-    switch (shell) {
-      case "fish":
-        shellProfilePath = `${home}/.config/fish/config.fish`;
-        break;
-      case "zsh":
-        shellProfilePath = `${home}/.zshrc`;
-        break;
-      default:
-        shellProfilePath = `${home}/.bashrc`;
-        break;
-    }
-
     const result = await asyncExec(
-      `source ${shellProfilePath} > /dev/null 2>&1 && ${ruby} -rjson -e "puts JSON.dump(ENV.to_h)"`,
-      { shell, cwd: this.workingFolder }
+      // eslint-disable-next-line no-process-env
+      `${process.env.SHELL} -lic '${ruby} -rjson -e "puts JSON.dump(ENV.to_h)"'`,
```

Comment:
What's the `-c` flag? I don't see it listed for `zsh -h`.

Reply:
It's used to execute a single command.

Reply:
Ah, it's in the `man` page, I wonder why it's missing from the help.

```
> INVOCATION
       The following flags are interpreted by the shell when invoked to determine where the shell will read commands from:

       -c     Take the first argument as a command to execute, rather than reading commands from a script or standard input.  If any further arguments are given, the first one is assigned to $0, rather than being used as a
              positional parameter.
```

Is it safe to assume all three of these are universal to all the shells we support?

Reply:
Yes, the three flags exist in bash, zsh and fish (and they do the same thing).